### ISSUE TITLE
Adding saving functionality

### ISSUE THREAD

**Issue by MHX792** [JanGorman/Agrume#24]
Great library! Any plans on adding a way to save the images to the library? Maybe on Longpress?

**Comment by JanGorman** [comment]
Yeah, something to consider. I'll think of a flexible way to add custom gesture support

**Comment by MHX792** [comment]
I added it to `AgrumeCell` like so:
```
    // MARK: - Saving to Photo Library with Long Press
    
    private lazy var longPressGesture: UILongPressGestureRecognizer = {
        let longPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(handleLongPress))
        longPressGesture.requireGestureRecognizerToFail(self.doubleTapGesture)
        longPressGesture.delegate = self
        return longPressGesture
    }()
    
    func handleLongPress(sender: UILongPressGestureRecognizer) {
        if (sender.state == UIGestureRecognizerState.Ended) {
            print("Long press Ended")
        } else if (sender.state == UIGestureRecognizerState.Began) {
            presentActionSheet()
        }
    }
    
    func presentActionSheet() {
        let action = UIActionSheet(title: nil, delegate: self, cancelButtonTitle: "Cancel", destructiveButtonTitle: nil, otherButtonTitles: "Save to Photo Library")
        action.showInView(self)
    }
    
    func actionSheet(actionSheet: UIActionSheet, clickedButtonAtIndex buttonIndex: Int) {
        if buttonIndex != actionSheet.cancelButtonIndex {
            UIImageWriteToSavedPhotosAlbum(imageView.image!, self, "image:didFinishSavingWithError:contextInfo:", nil)
        }
    }
    
    func image(image: UIImage, didFinishSavingWithError error: NSError?, contextInfo:UnsafePointer<Void>) {
        if error == nil {
            print("photo saved to library")
        } else {
            pront("saving error")
        }
    }

```

**Comment by MHX792** [comment]
Another more nifty solution is adding a toolbar and an action share button:

```
    override public func viewDidLoad() {
        super.viewDidLoad()
        // ...

        let toolbar = UIToolbar()
        toolbar.frame = CGRectMake(0, self.view.frame.size.height - 46, self.view.frame.size.width, 46)
        toolbar.sizeToFit()
        
        let actionButton = UIBarButtonItem(barButtonSystemItem: .Action, target: self, action: "didPressActionButton")
        actionButton.tintColor = UIColor.whiteColor()
        
        toolbar.setItems([actionButton], animated: true)
        toolbar.setBackgroundImage(UIImage(), forToolbarPosition: UIBarPosition.Any, barMetrics: UIBarMetrics.Default)
        toolbar.setShadowImage(UIImage(), forToolbarPosition: UIBarPosition.Any)
        self.view.addSubview(toolbar)

    }
    
    func didPressActionButton() {
        delegate?.promptToSaveImage()
        // The delegate is the AgrumeCell showing the image. 
        // Since I only show an image at a time there are no conflicts.
        // For multiple images this won't work.
    }
```


The result:

![simulator screen shot 25 jul 2016 11 05 00](https://cloud.githubusercontent.com/assets/12123471/17096029/b3ec5974-5257-11e6-9926-9f2a1ab52f98.png)

**Comment by mdilaveroglu** [comment]
+1

**Comment by Alexookah** [comment]
https://github.com/JanGorman/Agrume/pull/151
this does something similar with longPressGesture

**Comment by JanGorman** [comment]
Solved with #151
